File: native-android/modulo-02/clase-03/backend-app-taxi/src/features/passengers/dto/passenger.dto.ts
import { ApiProperty, ApiPropertyOptional } from "@nestjs/swagger";
import { PassengerStatus } from "../enum/passenger-status.enum";

/* -------------------------------------------------------
   DTO de lectura para Passenger (1:1 con la entidad)
   - Usado típicamente como respuesta en controladores
-------------------------------------------------------- */
export class PassengerDto {
    @ApiProperty({ description: "Passenger ID (UUID v4)", format: "uuid" })
    id!: string;

    @ApiProperty({
        description:
            "Phone number in E.164 without plus sign or symbols (e.g., 51987654321)",
        maxLength: 20,
    })
    phoneNumber!: string;

    @ApiPropertyOptional({ description: "Given name" })
    givenName?: string | null;

    @ApiPropertyOptional({ description: "Family name" })
    familyName?: string | null;

    @ApiPropertyOptional({ description: "Email address" })
    email?: string | null;

    @ApiPropertyOptional({ description: "Photo URL" })
    photoUrl?: string | null;

    @ApiProperty({
        description: "Business status",
        enum: PassengerStatus,
        enumName: "PassengerStatus",
        default: PassengerStatus.ACTIVE,
    })
    status!: PassengerStatus;
}

File: native-android/proyecto-final/app-taxi/android-passenger-app-taxi/app/src/main/java/com/example/android_passenger/commons/data/local/dto/PassengerDto.kt
package com.example.android_passenger.commons.data.local.dto

import com.example.android_passenger.commons.domain.model.Passenger

data class PassengerDto(
    val id: String,
    val phoneNumber: String,
    val givenName: String?,
    val familyName: String?,
    val email: String?,
    val photoUrl: String?,
    val status: String
)

fun PassengerDto.toDomain(): Passenger = Passenger(
    id = id,
    phoneNumber = phoneNumber,
    givenName = givenName,
    familyName = familyName,
    email = email,
    photoUrl = photoUrl,
    status = status
)